File: backend/API/aspnetcore/lcpsnapi/Classes/Media.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using static lcpsnapi.Classes.Enums;

namespace lcpsnapi.Classes
{
    public class Media
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public int? MediaId { get; set; }
        public string? Title { get; set; }
        public string? Url { get; set; }
        public string? Type { get; set; }
        public string? Category { get; set; }
        public string? Cover { get; set; }
        public string? Desc { get; set; }
        public Status? Status { get; set; }
        public Privacy? Privacy { get; set; }
        public bool? IsFeatured { get; set; }
        public string? DateUploaded { get; set; }
        public string? DateModified { get; set; }
        public string? DateDeleted { get; set; }
        public int? UsersId { get; set; }
    }
}

File: frontend/src/classes/media.ts
export class Videos {
    videosId?: number;
    title?: string;
    url?: string;
    type?: string;
    category?: string;
    cover?: string;
    desc?: string;
    status?: Status;
    privacy?: Privacy;
    isFeatured?: boolean;
    dateUploaded?: string | Date;
    dateModified?: string | Date;
    dateDeleted?: string | Date;
    usersId?: number;
}

export class Images {
    imagesId?: number;
    title?: string;
    url?: string;
    type?: string;
    category?: string;
    cover?: string;
    desc?: string;
    status?: Status;
    privacy?: Privacy;
    isFeatured?: boolean;
    dateUploaded?: string | Date;
    dateModified?: string | Date;
    dateDeleted?: string | Date;
    usersId?: number;
}

export class Files {
    filesId?: number;
    title?: string;
    url?: string;
    type?: string;
    category?: string;
    cover?: string;
    desc?: string;
    status?: Status;
    privacy?: Privacy;
    isFeatured?: boolean;
    dateUploaded?: string | Date;
    dateModified?: string | Date;
    dateDeleted?: string | Date;
    usersId?: number;
}

type Status = "published" | "updated" | "deleted" | "locked";
type Privacy = "public" | "private" | "only_friends_and_followers" | "only_friends" | "only_followers";